go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -65,14 +70,14 @@ type NodeController struct {
 //  2. Setting back annotations about its VTEP and gateway MAC address to its own object
 func NewNode(kube kube.Interface, nodeName string, stopChan <-chan struct{}) (*NodeController, error) {
 	node := &NodeController{
-		kube:        kube,
-		nodeName:    nodeName,
-		tunMap:      make(map[string]string),
-		tunMapMutex: sync.Mutex{},
-		flowCache:   make(map[string]*flowCacheEntry),
-		flowMutex:   sync.Mutex{},
-		flowChan:    make(chan struct{}, 1),
-		stopChan:    stopChan,
+		kube:       kube,
+		nodeName:   nodeName,
```

Comment:
initialize wf to emtpy struct here, otherwise the tests panic with a nil dereference

Reply:
I know you didn't fix node_linux_test as it was late for you, I spotted these changes since startNodeWatch now pulls the watchFactory from the node struct:

http://paste.openstack.org/show/793650/

Reply:
Thanks applied and repushed.